Website/src/rendering/src/redux/ocCurrentCart/index.ts

```diff
@@ -233,8 +238,29 @@ export const retrieveCart = createOcAsyncThunk<RequiredDeep<DOrderWorksheet> | u
       ) {
         ThunkAPI.dispatch(retrievePayments(existingOrder.ID));
       }
-      ThunkAPI.dispatch(retrievePromotions(existingOrder.ID));
+      const existingOrderPromos = (await ThunkAPI.dispatch(retrievePromotions(existingOrder.ID)))
```

Comment:
Have you tried to put this code in the `mergeAnonOrder` function? It seems that its job is to merge the orders, thus it seems to be the logical place to put this new code.